File: DevTools/Builder/src/main/java/info/smart_tools/smartactors/builder/jcommander/Arguments.java
package info.smart_tools.smartactors.builder.jcommander;

import com.beust.jcommander.Parameter;
import info.smart_tools.simpleactors.commons.IArguments;

public class Arguments implements IArguments {

    @Parameter(
            names = {"command", "cmd", "c", "-command", "-cmd", "-c"},
            description = "Command. Available values: 'build'. Default value is 'build'."
    )
    private String command = "build";

    @Parameter(
            names = {"-featurePath", "featurePath", "-fp"},
            description = "Path to the feature archive for upload, can be .zip or .jar"
    )
    private String featurePath;

    @Override
    public String getCommand() {
        return command;
    }

    public void setCommand(String command) {
        this.command = command;
    }

    public String getFeaturePath() {
        return featurePath;
    }

    public void setFeaturePath(String featurePath) {
        this.featurePath = featurePath;
    }
}

File: DevTools/SmartActorsIdeaPlugin/src/main/kotlin/info/smart_tools/smartactors/devtools/builder/tool/Arguments.kt
package info.smart_tools.smartactors.devtools.builder.tool

data class Arguments(
    val featurePath: String
) {
    fun toArray(): Array<out String> {
        val args = ArrayList<String>()
        args.add("-featurePath")
        args.add(featurePath)

        return args.toTypedArray()
    }
}
